.github/workflows/pipeline-perf-test.yml

```diff
@@ -0,0 +1,43 @@
+name: Pipeline Performance Tests
+
+on:
+    push:
+        branches: [ main ]
+    pull_request:
+        branches: [ main ]
+
+jobs:
+    pipeline-perf-test:
+        if: ${{ contains(github.event.pull_request.labels.*.name, 'performance') }}
+        runs-on: ubuntu-latest
+        steps:
+        - name: Harden the runner (Audit all outbound calls)
+          uses: step-security/harden-runner@c6295a65d1254861815972266d5933fd6e532bdf # v2.11.1
+          with:
+            egress-policy: audit
+
+        - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        
+        - name: Set up Python
+          uses: actions/setup-python@v4
```

Comment:
didn't use the pinned version with commit hash, as I hope https://github.com/open-telemetry/otel-arrow/pull/379 will automatically for us!